lib/file-helper.js

```diff
@@ -78,13 +78,13 @@ exports.getFingerprint = function (file) {
     // with a forward slashes.
     const relativeFile = path.join(filePath, file).replace(/\\/g, '/')
 
-    if (fingerprints.hasOwnProperty(relativeFile)) {
+    if (Object.prototype.hasOwnProperty.call(fingerprints, relativeFile)) {
```

Comment:
I've had a quick look and `fingerprints` is a certified 100% plain object™ 😆 

(No prototype, no iterator etc)

Easier to read as a quick truthy check instead?

```suggestion
    if (fingerprints[relativeFile]) {
```

Otherwise too might be easier to follow a single return at the end of the function pattern?

For a lot of the relative paths we already have a truthy `fingerprints[file]` 🎉 

```js
// Grab fingerprint array from the template context
const filePath = this.lookup('path')
const fingerprints = this.lookup('fingerprint')

// If that fails, and we know the path of the current file, look for a
// fingerprinted asset relative to the current file (e.g. `../foo.css`)
//
// We only know the path of the current file when we're compiling the layout –
// calls to this function with a relative path will fail if made from the
// source files themselves.
if (!fingerprints[file] && filePath) {
  // Use path.join to correctly join, but metalsmith-fingerprint-ignore
  // always expects forward slashes, so replace any backslashes (Windows)
  // with a forward slashes.
  file = path.join(filePath, file).replace(/\\/g, '/')
}

// The thrown error will stop the build, but not provide any useful output,
// so we have to console.log as well.
if (!fingerprints[file]) {
  console.log(`Could not find fingerprint for file ${file}`)
  throw new Error(`Could not find fingerprint for file ${file}`)
}

// Look for a fingerprinted asset at this path relative to the site root
return '/' + fingerprints[file]
```

Reply:
Awesome! Refactored.